examples/nextjs-example/pages/index.tsx

```diff
@@ -51,12 +51,26 @@ export default function Home() {
   return (
     <KnockProvider
       userId={userId as any}
+      userToken={userToken}
       apiKey={process.env.NEXT_PUBLIC_KNOCK_PUBLIC_API_KEY!}
       host={process.env.NEXT_PUBLIC_KNOCK_HOST}
+      authenticateOptions={{
+        onUserTokenExpiring: async () => {
+          // Refresh the user token 1s before it expires
+          const res = await fetch(`/api/auth?id=${userId}`);
+          const json = await res.json();
+          return json.userToken;
+        },
+        timeBeforeExpirationInMs: 500,
+      }}
```

Comment:
The only thing I was thinking if we should make these top level props instead of under an object. Might be cleaner?